base/stats.go

```diff
@@ -506,6 +509,49 @@ func (s *SgwFloatStat) Value() float64 {
 	return math.Float64frombits(atomic.LoadUint64(&s.Val))
 }
 
+// SgwDurStat is a wrapper around SgwStat for reporting time duration stats.
+type SgwDurStat struct {
+	SgwStat           // SGW stats for sending metrics to Prometheus.
+	Val     time.Time // Start time to calculate the time duration.
+}
+
+// NewDurStat creates a new collector for time duration metric, registers it with the
+// Prometheus's DefaultRegisterer and returns the collector. It panics if any error
+// occurs while registering the collector Prometheus registry.
+func NewDurStat(subsystem string, key string, labelKeys []string, labelVals []string,
+	statValueType prometheus.ValueType, initialValue time.Time) *SgwDurStat {
+	stat := &SgwDurStat{
+		SgwStat: *newSGWStat(subsystem, key, labelKeys, labelVals, statValueType),
+		Val:     initialValue,
+	}
+	prometheus.MustRegister(stat)
+	return stat
+}
+
+// Describe sends the super-set of all possible descriptors of metrics
+// collected by duration stats Collector to the provided channel and
+// returns once the last descriptor has been sent. Not implemented yet.
+func (s *SgwDurStat) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
Would be better to tailor these comments to us. This seems lifted from the Prometheus source code. In our case its just the descriptor for the single stat. Things like super-set could be confusing